src/shogun/CMakeLists.txt

```diff
@@ -63,15 +64,16 @@ FOREACH(template ${LIBSHOGUN_SRC_TMP})
 
     ADD_CUSTOM_COMMAND(OUTPUT ${generated_cpp}
         COMMAND ${CMAKE_COMMAND} -E copy_if_different ${template} ${generated_cpp}
-        COMMAND ${PYTHON_EXECUTABLE} ${generator_script} ${generated_cpp} -in ${CMAKE_CURRENT_BINARY_DIR}/headers_list.txt
+        COMMAND ${PYTHON_EXECUTABLE} ${generator_script} ${generated_cpp} ${generated_json}
```

Comment:
I'm not sure I'm following here, currently the headers list is given by `{$LIBSHOGUN_HEADERS}`, instead of passing this list as an argument to the script (or `headers_list.txt` in case of MSVC) you want to pass the ctags file and in the script having the function `parseCtags()` from [generate.py](https://github.com/shogun-toolbox/shogun/blob/feature/cereal/examples/meta/generator/generate.py#L20), then extract all the headers from this dictionary, something like:
```
tags=parseCtags(arg)
HEADERS=filter(lambda x: x.endswith('.h'), tags.values())
```
And the remaining part of the script stays the same, correct?